### PR TITLE
[DP 62] - Added lint

### PR DESCRIPTION
**Pull request by igornaydenov246**

### CODE REVIEW

File: src/components/LmsConfigurations/index.jsx [hunk 1]
```diff
@@ -33,32 +36,52 @@ class LmsConfigurations extends React.Component {
...
+              ? responses[3].value.data.results[0]
+              : null,
+          degreedConfig:
+            responses[4].status === 'fulfilled'
+              ? responses[4].value.data.results[0]
+              : null,
+          cornerstoneConfig:
+            responses[5].status === 'fulfilled'
+              ? responses[5].value.data.results[0]
+              : null,
```
Comment (Eugene-Kuzmenko) [inline src/components/LmsConfigurations/index.jsx:84]: This is repeated piece of code. We can use iteration for this. Even write a helper function
```suggestion
     ...mapPromiseSettlingResults(results, ({ data }) => data.results[0], [
        'moodleConfig',
        'canvasConfig',
        'blackboardConfig',
        'sapsfConfig',
        'degreedConfig',
        'cornerstoneConfig',
     ])
```

Comment (Eugene-Kuzmenko) [reply src/components/LmsConfigurations/index.jsx:84]:
```javascript
function mapPromiseSettlingResults(results, getValue, orderedKeys) {
  const mapped = {};
  orderedKeys.forEach((key, i) => {
     const { status, value } = results[i];
     mapped[key] = status === 'fulfilled' ? getValue(value) : null;
  });
  return mapped;
}
```

File: src/components/SamlProviderConfiguration/SamlProviderDataForm.jsx [hunk 2]
```diff
@@ -28,10 +31,13 @@ class SamlProviderDataForm extends React.Component {
    */
   validateProviderDataForm = (formData, requiredFields) => {
     const invalidFields = requiredFields
-      .filter(field => !formData.get(field))
-      .reduce((prevFields, currField) => ({ ...prevFields, [currField]: true }), {});
+      .filter((field) => !formData.get(field))
+      .reduce(
+        (prevFields, currField) => ({ ...prevFields, [currField]: true }),
+        {},
+      );
```
Comment (Eugene-Kuzmenko) [inline src/components/SamlProviderConfiguration/SamlProviderDataForm.jsx:38]: That's O(n^2). There is a better way
```suggestion
     const invalidFields = {};
     for (let field of requiredFields) {
        if (!formData.get(field)) invalidFields[field] = true;
     }
```

File: package.json [hunk 3]
```diff
@@ -31,6 +37,7 @@
     "@fortawesome/free-regular-svg-icons": "5.15.3",
     "@fortawesome/free-solid-svg-icons": "5.15.3",
     "@fortawesome/react-fontawesome": "0.1.14",
+    "@reustleco/dojo-frontend-common": "1.1.0-alpha.3 ",
```
Comment (Alain1405) [inline package.json:40]: Use the latest version `1.1.0`

**Review by Alain1405 — COMMENTED** [review]
Just one comment about the version of common package but I will leave the last word to @Eugene-Kuzmenko

File: src/components/SamlProviderConfiguration/index.jsx [hunk 4]
```diff
@@ -17,28 +17,40 @@ import { configuration } from '../../config';
...
+      .then((responses) => {
+        this.setState({
+          providerConfig:
+            responses[0].status === 'fulfilled'
+              ? responses[0].value.data.results[0]
+              : undefined,
+          providerData:
+            responses[1].status === 'fulfilled'
+              ? responses[1].value.data.results[0]
+              : undefined,
```
Comment (Eugene-Kuzmenko) [inline src/components/SamlProviderConfiguration/index.jsx:46]:
```suggestion
          ...mapSettlingResults(results, ({ data }) => data.results[0], ['providerConfig', 'providerData']),
```

**Review by Eugene-Kuzmenko — CHANGES_REQUESTED** [review]

**Pull request merged**